sources/models/shared-defaults/aws-creds.toml

```diff
@@ -11,6 +11,9 @@ restart-commands = []
 [metadata.settings.aws]
 affected-services = ["aws"]
 
+[metadata.settings.aws.profile]
+affected-services = ["kubernetes"]
```

Comment:
I'd expect that this needs to go in a k8s-specific model file, since `aws-ecs-1` will include this but will not have a "kubernetes" service in the model.

Reply:
This is only pulled in to k8s specific models:

```sh
$ find ./ -name *aws-creds.toml
./sources/models/shared-defaults/aws-creds.toml
./sources/models/src/aws-k8s-1.26-nvidia/defaults.d/40-aws-creds.toml
./sources/models/src/aws-k8s-1.22-nvidia/defaults.d/40-aws-creds.toml
./sources/models/src/aws-k8s-1.26/defaults.d/40-aws-creds.toml
./sources/models/src/vmware-k8s-1.22/defaults.d/40-aws-creds.toml
./sources/models/src/metal-k8s-1.22/defaults.d/40-aws-creds.toml
./sources/models/src/vmware-k8s-1.26/defaults.d/40-aws-creds.toml
./sources/models/src/aws-k8s-1.22/defaults.d/40-aws-creds.toml
./sources/models/src/metal-k8s-1.26/defaults.d/40-aws-creds.toml
```

Reply:
That's a separate bug then since it means the ECS variant will accept these settings in the API but treat them as a no-op - not rendering any files. Which I suppose is OK since nothing uses those settings, but is still surprising.

Maybe rename this file so it's clear that it's k8s-specific? Then linking it into the ECS models in the future might be flagged by a reviewer.

Reply:
Could it be possible we may need to use AWS credentials for ECS or other platforms in the future though?

The only k8s-specific part about this is the service restart. Would it be possible to have:

```toml
[metadata.settings.aws.profile]
affected-services = ["kubernetes"]
```

In another model that would only be pulled in to k8s variants, leaving this open to being able to be used in non-k8s variants for other use cases?

Reply:
Yes, if we put this in one of the k8s templates then that should be fine.